### PR TITLE
[core] fix(Slider, MultiSlider): improve a11y markup

### PR DESCRIPTION
**Pull request by bvandrc**
#### Checklist

- [ ] Includes tests
- [X] Update documentation

#### Changes proposed in this pull request:

improve slider a11y per this official ARIA example: https://www.w3.org/WAI/ARIA/apg/example-index/slider/slider-multithumb.html

### CODE REVIEW

File: packages/core/src/components/slider/handle.tsx [hunk 1]
```diff
@@ -80,6 +80,12 @@ export class Handle extends AbstractPureComponent2<IInternalHandleProps, IHandle
                 ref={this.refHandlers.handle}
                 style={this.getStyleProperties()}
                 tabIndex={0}
+                role="slider"
+                aria-valuemin={min}
+                aria-valuemax={max}
+                aria-valuenow={value}
```
Comment (bvandrc) [inline packages/core/src/components/slider/handle.tsx:87]: `value` gives the current value of this Handle, correct?

Comment (adidahiya) [reply packages/core/src/components/slider/handle.tsx:87]: yes

File: packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx [hunk 2]
```diff
@@ -58,6 +58,7 @@ export class CollapsibleListExample extends React.PureComponent<IExampleProps, I
                     onChange={this.handleChangeCount}
                     showTrackFill={false}
                     value={this.state.visibleItemCount}
+                    handleAriaLabel="number of visible items"
```
Comment (adidahiya) [inline packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx]: This slider (and the other sliders in the docs examples) may be better labeled by their visible label text inside the preceding `<Label>` elements... something like:

```
<Label id={labelId}>Visible items</Label>
<Slider handleProps={{ ariaLabelledBy: labelId }}>
```

what do you think?

File: packages/core/src/components/slider/handleProps.tsx [hunk 3]
```diff
@@ -89,4 +89,7 @@ export interface IHandleProps extends Props {
      * @default "full"
      */
     type?: HandleType;
+
+    /** Applied to `aria-label` prop of the handle `span` element */
```
Comment (adidahiya) [inline packages/core/src/components/slider/handleProps.tsx]:
```suggestion
    /** Accessible label text for this handle element. */
```

File: packages/core/src/components/slider/slider.tsx [hunk 4]
```diff
@@ -44,6 +44,9 @@ export interface ISliderProps extends ISliderBaseProps {
 
     /** Callback invoked when the handle is released. */
     onRelease?(value: number): void;
+
+    /** Applied to `aria-label` prop of the slider Handle. */
+    handleAriaLabel?: string;
```
Comment (adidahiya) [inline packages/core/src/components/slider/slider.tsx]: According to the [MDN ARIA: slider role docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/slider_role), we should also support `aria-labelledby` in addition to `aria-label`... I'm thinking the API would look better if we forwarded a subset of handle props from Slider...

```suggestion
    /** Subset of handle props to apply to the child Handle. */
    handleProps?: Pick<HandleProps, "ariaLabel" | "ariaLabelledBy">;
```

File: packages/core/src/components/slider/rangeSlider.tsx [hunk 5]
```diff
@@ -44,6 +44,9 @@ export interface IRangeSliderProps extends ISliderBaseProps {
 
     /** Callback invoked when a handle is released. */
     onRelease?(value: NumberRange): void;
+
+    /** Applied to `aria-label` props of the slider Handles */
+    handleAriaLabels?: { startAriaLabel: string; endAriaLabel: string };
```
Comment (adidahiya) [inline packages/core/src/components/slider/rangeSlider.tsx]: following my comment on `slider.tsx`... my API suggestion here would be

```suggestion
    /** Subset of handle props to apply to the start and end Handles. */
    handleProps?: {
    	start: Pick<HandleProps, "ariaLabel" | "ariaLabelledBy">;
    	end: Pick<HandleProps, "ariaLabel" | "ariaLabelledBy">;
    };
```

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx (hunk 2 above)
Comment (bvandrc) [reply packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx]: yep, I agree. Implemented.

File: packages/core/src/components/slider/slider.tsx (hunk 4 above)
Comment (bvandrc) [reply packages/core/src/components/slider/slider.tsx]: yep you're right, we should support `aria-labelledby`. Did it a little differently, see changes

File: packages/core/src/components/slider/rangeSlider.tsx (hunk 5 above)
Comment (bvandrc) [reply packages/core/src/components/slider/rangeSlider.tsx]: yep, see new changes

File: packages/core/src/components/slider/handleProps.tsx [hunk 6]
```diff
@@ -47,6 +47,8 @@ export const HandleInteractionKind = {
 // eslint-disable-next-line @typescript-eslint/no-redeclare
 export type HandleInteractionKind = typeof HandleInteractionKind[keyof typeof HandleInteractionKind];
 
+export type HandleHtmlProps = Pick<React.HTMLProps<HTMLSpanElement>, "aria-label" | "aria-labelledby">;
```
Comment (bvandrc) [inline packages/core/src/components/slider/handleProps.tsx:50]: do we actually want to `Pick` these specific props or should we just let any `span` props be passed? (Should this just be `React.HTMLProps<HTMLSpanElement>`?)

File: packages/core/src/components/slider/handleProps.tsx [hunk 7]
```diff
@@ -89,4 +91,9 @@ export interface IHandleProps extends Props {
      * @default "full"
      */
     type?: HandleType;
+
+    /**
+     * HTML props to apply to the Handle
+     */
+    handleHtmlProps?: HandleHtmlProps;
```
Comment (adidahiya) [inline packages/core/src/components/slider/handleProps.tsx]:
```suggestion
     * A limited subset of HTML props to apply to the rendered `<span>` element.
     */
    htmlProps?: HandleHtmlProps;
```

**Review by adidahiya — CHANGES_REQUESTED** [review]
Looks good! Just one small edit. `<Handle handleHtmlProps={...}>` feels verbose/redundant so I prefer `<Handle htmlProps={...}>`.

`<Slider handleHtmlProps={...}>` is sufficiently verbose, though, let's keep that prop name.

File: packages/core/src/components/slider/handleProps.tsx (hunk 6 above)
Comment (adidahiya) [reply packages/core/src/components/slider/handleProps.tsx:50]: Let's just pick these two for now. We can add more later if necessary

Comment (bvandrc) [reply packages/core/src/components/slider/handleProps.tsx:50]: sounds good

File: packages/core/src/components/slider/handleProps.tsx (hunk 7 above)
Comment (bvandrc) [reply packages/core/src/components/slider/handleProps.tsx]: done, changed

File: packages/core/src/components/slider/rangeSlider.tsx [hunk 8]
```diff
@@ -56,11 +60,20 @@ export class RangeSlider extends AbstractPureComponent2<RangeSliderProps> {
...
+        const { value, handleHtmlProps, ...props } = this.props;
         return (
             <MultiSlider {...props}>
-                <MultiSlider.Handle value={value![RangeIndex.START]} type="start" intentAfter={props.intent} />
-                <MultiSlider.Handle value={value![RangeIndex.END]} type="end" />
+                <MultiSlider.Handle
+                    value={value![RangeIndex.START]}
+                    type="start"
+                    intentAfter={props.intent}
+                    htmlProps={handleHtmlProps ? handleHtmlProps.start : undefined}
```
Comment (adidahiya) [inline packages/core/src/components/slider/rangeSlider.tsx]:
```suggestion
                    htmlProps={handleHtmlProps?.start}
```

File: packages/core/src/components/slider/rangeSlider.tsx [hunk 9]
```diff
@@ -56,11 +60,20 @@ export class RangeSlider extends AbstractPureComponent2<RangeSliderProps> {
...
+                />
+                <MultiSlider.Handle
+                    value={value![RangeIndex.END]}
+                    type="end"
+                    htmlProps={handleHtmlProps ? handleHtmlProps.end : undefined}
```
Comment (adidahiya) [inline packages/core/src/components/slider/rangeSlider.tsx]:
```suggestion
                    htmlProps={handleHtmlProps?.end}
```

File: packages/core/src/components/slider/rangeSlider.tsx [hunk 10]
```diff
@@ -44,6 +45,9 @@ export interface IRangeSliderProps extends ISliderBaseProps {
 
     /** Callback invoked when a handle is released. */
     onRelease?(value: NumberRange): void;
+
+    /** HTML props to apply to the slider Handles */
+    handleHtmlProps?: { start: HandleHtmlProps; end: HandleHtmlProps };
```
Comment (adidahiya) [inline packages/core/src/components/slider/rangeSlider.tsx]: I just realized that these should probably be optional, so you can only set one of them if you want:

```suggestion
    handleHtmlProps?: { start?: HandleHtmlProps; end?: HandleHtmlProps };
```

**Review by adidahiya — CHANGES_REQUESTED** [review]

**Review by adidahiya — APPROVED** [review]

**Pull request merged**